docs/content/3.composables/1.useCldImageUrl.md

```diff
@@ -0,0 +1,173 @@
+# `useCldImageUrl`
+
+---
+
+This composable is using [@cloudinary-util/url-loader](https://github.com/colbyfayock/cloudinary-util/tree/main/packages/url-loader) under the hood to generate the Cloudinary URL with certain optimizations.
+
+## Usage
+
+The composable allows you to pass optimization options and the src of the image and in return get a full cloudinary url with the optimized images. Then, you can use this url any way you like (for example in the custom image component).
+
+```vue
+<script lang="ts" setup>
+const { url } = useCldImageUrl({
+  options: {
+    src: "/cld-sample-5.jpg",
+    tint: "40:253f8c",
+    underlay: "/cld-sample-4.jpg",
+    overlays: [
+      {
+        position: {
+          gravity: "north",
+          y: 60,
+        },
+        text: {
+          color: "rgb:52a4ff80",
+          fontFamily: "Source Sans Pro",
+          fontSize: 320,
+          fontWeight: "black",
+          text: "MUSIC",
+          letterSpacing: -10,
+          lineSpacing: -100,
+          stroke: true,
+          border: "20px_solid_rgb:2d0eff99",
+        },
+      },
+    ],
+  },
+});
+</script>
+```
+
+## Configuration
+
+Apart from the image options, `useCldImageUrl` composable accepts two additional parameters; `config` and `analytics`.
```

Comment:
imo config and analytics aren't really designed to be exposed, which is why i made them secondary undocumented options (not sure if not documenting was the right call or not, open to feedback)

config really has 1 objective for 99% of use cases and thats to pass in the cloud name. that should be done on behalf of the user, similar to CldImage

analytics is similar, but what that does is pass an internal flag to the Cloudinary URL builder that allows us to "track" usage via HTTP requests. it just lets us see how many people are using X library via a URL parameter (ex: myimage?_a=X)

the user shouldnt configure that

Reply:
similar to previous comment, you can see how im setting it up and abstracting that away iwthin getCldImageUrl https://github.com/colbyfayock/next-cloudinary/blob/main/next-cloudinary/src/helpers/getCldImageUrl.ts#L20

Reply:
https://github.com/nuxt-community/cloudinary-module/pull/111#discussion_r1198966866